tests/core/custom_types/test_coin.py

```diff
@@ -76,34 +76,34 @@ def test_construction() -> None:
     H1 = b"a" * 32
     H2 = b"b" * 32
 
-    with pytest.raises(OverflowError, match="int too big to convert"):
+    with pytest.raises(ValueError, match="Value 18446744073709551616 does not fit into uint64"):
         # overflow
-        Coin(H1, H2, 0x10000000000000000)
+        Coin(H1, H2, uint64(0x10000000000000000))
 
-    with pytest.raises(OverflowError, match="can't convert negative int to unsigned"):
+    with pytest.raises(ValueError, match="Value -1 does not fit into uint64"):
         # overflow
-        Coin(H1, H2, -1)
+        Coin(H1, H2, uint64(-1))
```

Comment:
It feels like we're changing the test here.  We're now testing the `uint64()` constructor instead of the `Coin()` constructor, right?  This can then just be deleted.  Or, we should test the `Coin()` constructor still if we want it to raise for either the wrong type (`int` not `uint64`) or out of bounds values.  But...  that would belong in `chia_rs`, right?

Reply:
yeah. maybe it would be better to make a mypy exception to allow passing in the wrong type

Reply:
If `Coin()` still checks then yeah, do that.  Though again at this point that should be tested in `chia_rs`.